sale_order_project/models/sale.py

```diff
@@ -1,64 +1,36 @@
 # -*- coding: utf-8 -*-
-###############################################################################
-#
-#   Module for OpenERP
-#   Copyright (C) 2014 Akretion (http://www.akretion.com).
-#   Copyright (C) 2010-2013 Akretion LDTA (<http://www.akretion.com>)
-#   @author Sébastien BEAU <sebastien.beau@akretion.com>
-#   @author Benoît GUILLOT <benoit.guillot@akretion.com>
-#
-#   This program is free software: you can redistribute it and/or modify
-#   it under the terms of the GNU Affero General Public License as
-#   published by the Free Software Foundation, either version 3 of the
-#   License, or (at your option) any later version.
-#
-#   This program is distributed in the hope that it will be useful,
-#   but WITHOUT ANY WARRANTY; without even the implied warranty of
-#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#   GNU Affero General Public License for more details.
-#
-#   You should have received a copy of the GNU Affero General Public License
-#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-###############################################################################
+# © 2014 Akretion - Sébastien BEAU <sebastien.beau@akretion.com>
+# © 2014 Akretion - Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp import api, fields, models
+from odoo import api, models, _
+from odoo.exceptions import Warning
 from datetime import date
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    @api.one
-    @api.depends('project_id')
-    def _compute_related_project_id(self):
-        self.related_project_id = (
-            self.project_id.use_tasks and
-            self.env['project.project'].search(
-                [('analytic_account_id', '=', self.project_id.id)],
-                limit=1)[:1])
-
-    related_project_id = fields.Many2one(
-        comodel_name='project.project', string='Project',
-        compute='_compute_related_project_id')
-
-    @api.model
-    def _prepare_project_vals(self, order):
+    def _prepare_project_vals(self):
         name = u" %s - %s - %s" % (
```

Comment:
Please add a `self.ensure_one()` (every function are `@api.multi` by default)

Reply:
Done, I forgot in the migration.